app/react/src/server/cra_config.js

```diff
@@ -0,0 +1,83 @@
+import semver from 'semver';
+
+let MiniCssExtractPlugin;
+let normalizeCondition;
+
+export function isReactScriptsInstalled() {
+  try {
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+    const reactScriptsJson = require('react-scripts/package.json');
+    if (semver.lt(reactScriptsJson.version, '2.0.0')) return false;
+
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies, prefer-destructuring
+    normalizeCondition = require('webpack/lib/RuleSet').normalizeCondition;
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+    MiniCssExtractPlugin = require('mini-css-extract-plugin');
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+export function getStyleRules(rules) {
+  // Extensions of style rules we're interested in
+  const extensions = ['.css', '.scss', '.sass', '.module.css', '.module.scss', '.module.sass'];
+
+  return rules.reduce((styleRules, rule) => {
+    if (rule.test && extensions.some(normalizeCondition(rule.test))) {
+      // If the base test is for styles, return early
+      return styleRules.concat(rule);
+    }
+
+    if (!rule.test && rule.oneOf) {
+      styleRules.push(...getStyleRules(rule.oneOf));
+    }
+
+    if (!rule.test && rule.rules) {
+      styleRules.push(...getStyleRules(rule.rules));
+    }
+
+    return styleRules;
+  }, []);
+}
+
+export function applyCRAWebpackConfig(baseConfig) {
+  // Remove style rules from baseConfig
+  const baseRulesExcludingStyles = baseConfig.module.rules.filter(
+    rule => !rule.test || !['.css', '.scss', '.sass'].some(normalizeCondition(rule.test))
+  );
+
+  //  Load create-react-app config
+  let craWebpackConfig;
+  if (baseConfig.mode === 'production') {
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+    craWebpackConfig = require('react-scripts/config/webpack.config.prod');
+  } else {
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+    craWebpackConfig = require('react-scripts/config/webpack.config.dev');
+  }
+
+  //  Concat will ensure rules is an array
+  const craStyleRules = getStyleRules([].concat(craWebpackConfig.module.rules));
+
+  //  Add css minification for production
+  const plugins = [...baseConfig.plugins];
+  if (baseConfig.mode === 'production')
+    plugins.push(
+      new MiniCssExtractPlugin({
```

Comment:
Do we really need this, though?

Reply:
Nope! I just thought minification would be good for production.

Also, if we don't want to include this plugin, then we have to filter out the related rule from all of the style rules. Adding the plugin was less work than removing it.
https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/webpack.config.prod.js#L65-L71

Reply:
@kylemh This preset checks for `react-scripts` greater than v2. So it would not do anything for CRAv1.